### ISSUE TITLE
Workers and `importScript`

### ISSUE THREAD

**Issue by chancancode** [embroider-build/embroider#1172]
Would we consider supporting workers (web workers and service workers) and [`importScripts`](https://developer.mozilla.org/en-US/docs/Web/API/WorkerGlobalScope/importScripts)? `importScripts` seems fairly easy, but I am not sure how to make workers work, other than just having a way to configure additional/multiple script entry points (which would also be useful for things like producing, say, a bookmarklet).

**Comment by ef4** [comment]
I don't think this requires any particular support from embroider. If the general-purpose javascript bundler we're using supports these features, so do we. 

A feature I would like to see that I don't think exists anywhere yet is a bundler that sees `navigator.serviceWorker.register('./foo.js', { type: 'module'})` as automatically defining a new entrypoint starting at foo.js in a worker context. In that sense, no embroider- or ember-specific configuration would be needed. All the intent is expressed in web-standards forms.

As a practical limitation right now, both webpack and typescript have a hard time managing a single configuration that spans both DOM and worker contexts, so I would tell people they'll be happier putting the worker code in a standalone build.

Webpack makes you choose between `target: 'web'` and `target: 'webworker'` and emits different module loader code for each case.

TypeScript has `compilerOptions.lib` that can include `"WebWorker"` or `"dom"` but not both because the types conflict with each other. So you need a separate tsconfig for each environment.

**Comment by ef4** [comment]
This week I actually happened to put together the shell of an application where the ember app builds with embroider and there's a separate build for a service worker, you can see it here: https://github.com/cardstack/runtime-spike

The ember app builds with embroider but doesn't have any special worker-related configuration. It launches the service worker [here](https://github.com/cardstack/runtime-spike/blob/3a1abf5e4bce0d9000546a549490925ff5f3e3cc/host/app/routes/application.ts#L6). 

And the service worker's main entrypoint is [here](https://github.com/cardstack/runtime-spike/blob/3a1abf5e4bce0d9000546a549490925ff5f3e3cc/worker/src/main.ts) and it has its own tsconfig and webpack config in that package.

**Comment by chancancode** [comment]
Thanks, I will take a look. Yeah, I was hoping that things like the worker registration and `importScripts` would automatically create new entry points, short of that I was hoping that there would at least be another natural way to enumerate additional entry points in the Embroider config or in package.json etc. You make a good point in though in this case that it probably works better as a separate workspace repo anyway. But I think a way to create entry points will still be useful for other cases like bookmarklets. (Maybe it already exists and I just missed it?)

**Comment by jrjohnson** [comment]
I don't think launching the worker in the application route works. If the first visit to the site is something like `/something/nested/334` then the worker won't be registered at `/` and will throw and error. I'd love to be wrong about this though. The spike is interesting, thanks for sharing. I'm trying to get the same functionality we have in ember-service-worker as possibly the last step to moving to an embroider build for our app, but there is a lot to digest.
